src/monattack.cpp

```diff
@@ -4217,10 +4295,81 @@ void mattack::suicide(monster *z, int index)
 {
     (void)index;
     Creature *target = z->attack_target();
-    if( target == nullptr || 
-        rl_dist( z->pos(), target->pos() ) > 2 || 
+    if( target == nullptr ||
+        rl_dist( z->pos(), target->pos() ) > 2 ||
         !z->sees( *target ) ) {
         return;
     }
     z->die(z);
 }
+
+void mattack::stretch_attack(monster *z, int index){
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if (target == nullptr || rl_dist(z->pos(), target->pos()) > 3 || !z->sees(*target)){
+        return;
+    }
+    int distance = rl_dist(z->pos(), target->pos());
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+    if (distance > 2 && distance <= 3){
+        z->moves -=100;
+        z->reset_special(index);
+        if ( foe != nullptr){
+            auto msg_type = foe == &g->u ? m_warning : m_info;
+            foe->add_msg_player_or_npc(msg_type, _("The %s thrusts its arm at you, stretching to reach you from afar"),
+                                                 _("The %s thrusts its arm at <npcname>, stretching to reach <npcname> from afar"),
+                z->name().c_str());
+        } else {
+            add_msg( _("The %s thrusts its arm at %s."),
+                        z->name().c_str(), target->disp_name().c_str());
+        }
+        int dodge_check = std::max(g->u.get_dodge() - rng(0, z->type->melee_skill), 0L);
+        if (rng(0, 10000) < 10000 / (1 + (99 * exp(-.6 * dodge_check))) || g->u.uncanny_dodge()) {
+            if ( foe != nullptr) {
+                auto msg_type = foe == &g->u ? m_warning : m_info;
+                foe->add_msg_player_or_npc(msg_type, _("You evade the stretched arm and it sails past you!"),
+                                                     _("<npcname> evades the stretched arm and it sails past <npcname>!"),
+                                                    z->name().c_str());
+                foe->practice( "dodge", z->type->melee_skill * 2 );
+                foe->ma_ondodge_effects();
+            } else {
+                add_msg( _("The %s evades the stretched arm and it sails past %s!"),
+                            target->disp_name().c_str());
+            }
+            //takes some time to retract the arm
+            z->moves -= 150;
```

Comment:
OK, this works for the arm (which, presumably, doesn't have any sensory inputs).